src/pages/Homepage/index.jsx

```diff
@@ -39,82 +31,69 @@ class Homepage extends Component {
-              <h1 className="text-center p-5">Giphy Search Engine</h1>
-            </div>
+  return (
+    <main className="homepage">
+      <div className="container">
+        <div className="row">
+          <div className="col-lg-12 col-sm-12">
+            <h1 className="text-center p-5">Giphy Search Engine</h1>
           </div>
-          <div className="row">
-            <div className="col-lg-6 col-sm-12 offset-lg-3 justify-content-center">
-              <form
-                action=""
-                className="search-form"
-                onSubmit={this.handleSubmit}
+        </div>
+        <div className="row">
+          <div className="col-lg-6 col-sm-12 offset-lg-3 justify-content-center">
+            <form action="" className="search-form" onSubmit={handleSubmit}>
+              <span className="error-message text-center">{errorMessage}</span>
+              <input
+                type="text"
+                name="keywords"
+                value={keywords}
+                onChange={handleChange}
+                className="search-input mb-3"
+                placeholder="Enter any keywords"
+                required
+              />
+              <button
+                className="btn-search"
+                type="submit"
+                onClick={handleSubmit}
               >
-                <span className="error-message text-center">
-                  {this.state.errorMessage}
-                </span>
-                <input
-                  type="text"
-                  name="keywords"
-                  value={this.state.keywords}
-                  onChange={this.handleChange}
-                  className="search-input mb-3"
-                  placeholder="Enter any keywords"
-                  required
-                />
-                <button
-                  className="btn-search"
-                  type="submit"
-                  onClick={this.handleSubmit}
-                >
-                  Search
-                </button>
-              </form>
-            </div>
+                Search
+              </button>
+            </form>
           </div>
+        </div>
 
-          <div className="row  mt-5 p-3">
-            {this.state.isLoading ? <Spinner /> : null}
+        <div className="row  mt-5 p-3">
+          {isLoading ? <Spinner /> : null}
 
-            {this.state.gifsData.map((val) => (
-              <div className="col-lg-3 col-sm-4" key={val.id}>
-                <Link
-                  to={{
-                    pathname: `gifdetails/${val.id}`,
-                    state: val,
-                  }}
-                >
-                  <GifCard gifImg={val.images.fixed_height_downsampled.url} />
-                </Link>
-                <Route
-                  path={`${this.props.match.path}/:id`}
-                  component={GifDetails}
-                />
-              </div>
-            ))}
-          </div>
+          {gifsData.map((val) => (
```

Comment:
instead of `val`, a better name might be `gif`. When you're mapping over things, use the singular noun of the items.

e.g 

`cars.map((car)=><div>{car.name}</div>)`


`persons.map((person)=><div>{person.name}</div>)`

Makes it easier to follow the code

Reply:
Done...